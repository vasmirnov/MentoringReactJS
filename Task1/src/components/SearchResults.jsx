import React from 'react';
import { Preview } from './Preview';

export class SearchResults extends React.PureComponent {

    render() {
        return (
            <div>
                {this.props.films.map((film) =>
                    <Preview key={film.id} film={film} selectFilmHandler={this.props.selectFilmHandler}/>
                )}   
            </div>
        )
    }
}