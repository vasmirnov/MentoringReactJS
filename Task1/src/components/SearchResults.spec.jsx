import React from 'react';
import renderer from 'react-test-renderer';
import { SearchResults } from './SearchResults';
import { Map } from 'immutable';

jest.disableAutomock();
jest.mock('./Preview', () => { return { Preview: 'Preview' } });

let props;

describe('SearchResults', () => {
    beforeEach(() => {
        props = {
            selectFilmHandler: () => { },
            films: [Map({ id: 1 }), Map({ id: 2 })]
        }
    });

    test('render correctly', () => {
        const instance = renderer.create(
            <SearchResults {...props} />
        );
        expect(instance.toJSON()).toMatchSnapshot();
    });
});