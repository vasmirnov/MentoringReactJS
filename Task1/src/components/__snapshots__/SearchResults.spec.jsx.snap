// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults render correctly 1`] = `
<div>
  <Preview
    film={
      Immutable.Map {
        "id": 1,
      }
    }
    selectFilmHandler={[Function]}
  />
  <Preview
    film={
      Immutable.Map {
        "id": 2,
      }
    }
    selectFilmHandler={[Function]}
  />
</div>
`;
